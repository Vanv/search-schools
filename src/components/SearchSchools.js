import React, { Component } from 'react';


class SearchSchools extends Component {
  render() {
    return(
      <div className="search-appointments row justify-content-center my-4">
        <div className="col-md-10">
          <div className="input-group">
            <input
              id="search-schools"
              type="text"
              className="form-control"
              placeholder="Search schools by suburb"
              onChange={e => this.props.searchSchools(e.target.value)}
            />
            <div className="input-group-append">
              <button
                type="button"
                className="btn btn-primary dropdown-toggle"
                data-toggle="dropdown"
                aria-haspopup="true"
                aria-expanded="false"
              >
                Sort by: <span className="caret" />
              </button>

              <div className="sort-menu dropdown-menu dropdown-menu-right">
                <button className={
                  'sort-by dropdown-item ' +
                  (this.props.orderBy === 'Suburb' ? 'active': '')
                }
                onClick={ e => this.props.changeOrder('Suburb', this.props.orderDir)}
                href="#">
                  Suburb
                </button>
                <button
                  className={
                   'sort-by dropdown-item ' +
                   (this.props.orderBy === 'SchoolName' ? 'active': '')
                 }
                 onClick={ e => this.props.changeOrder('SchoolName', this.props.orderDir)}
                 href="#">
                  School Name
                </button>
                <div role="separator" className="dropdown-divider" />
                <button  className={
                    'sort-by dropdown-item ' +
                    (this.props.orderDir === 'asc' ? 'active': '')
                  }
                  onClick={e => this.props.changeOrder(this.props.orderBy, 'asc')}
                  href="#">
                  Asc
                </button>
                <button
                    className={
                     'sort-by dropdown-item ' +
                     (this.props.orderDir === 'desc' ? 'active': '')
                   }
                   onClick={e => this.props.changeOrder(this.props.orderBy, 'desc')}
                   href="#">
                  Desc
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default SearchSchools;
