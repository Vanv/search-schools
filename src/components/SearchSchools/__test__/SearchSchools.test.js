import React from 'react';
import ReactDOM from 'react-dom';
import SearchSchools from './../SearchSchools';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

it("renders searchschools correctly", () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchSchools />, div);
})

it("matches snapshot", () => {
  const tree = renderer.create(<SearchSchools />).toJSON();
  expect(tree).toMatchSnapshot();
})
