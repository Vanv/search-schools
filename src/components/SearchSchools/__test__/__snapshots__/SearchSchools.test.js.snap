// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="search-schools row justify-content-center my-4"
  data-testid="searchschools"
>
  <div
    className="col-md-10"
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        id="search-schools"
        onChange={[Function]}
        placeholder="Search schools by suburb"
        type="text"
      />
      <div
        className="input-group-append"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          className="btn btn-primary dropdown-toggle"
          data-toggle="dropdown"
          type="button"
        >
          Sort by: 
          <span
            className="caret"
          />
        </button>
        <div
          className="sort-menu dropdown-menu dropdown-menu-right"
        >
          <button
            className="sort-by dropdown-item "
            href="#"
            onClick={[Function]}
          >
            Suburb
          </button>
          <button
            className="sort-by dropdown-item "
            href="#"
            onClick={[Function]}
          >
            School Name
          </button>
          <div
            className="dropdown-divider"
            role="separator"
          />
          <button
            className="sort-by dropdown-item "
            href="#"
            onClick={[Function]}
          >
            Asc
          </button>
          <button
            className="sort-by dropdown-item "
            href="#"
            onClick={[Function]}
          >
            Desc
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
